File: pkg/hashes/hashid.go
package hashes

import (
	"errors"
	"strconv"

	"github.com/speps/go-hashids/v2"
)

var (
	hasher *hashids.HashID
)

func init() {
	hd := hashids.NewData()
	hd.Salt = "azowernasdfoia"
	hd.MinLength = 5
	hasher, _ = hashids.NewWithData(hd)
}

func Encode(id int) (string, error) {
	return hasher.Encode([]int{id})
}

func Decode(hid string) (int, error) {
	ids, err := hasher.DecodeWithError(hid)
	if err != nil {
		return 0, err
	}
	if len(ids) < 1 {
		return 0, errors.New("invalid hash")
	}
	return ids[0], nil
}

func MaybeDecode(hid string) (int, error) {
	id, err := Decode(hid)
	if err != nil {
		id, err = strconv.Atoi(hid)
	}
	return id, err
}

File: packages/discovery-provider/plugins/notifications/src/utils/hashId.ts
import Hashids from 'hashids'

const HASH_SALT = 'azowernasdfoia'
const MIN_LENGTH = 5
const hashids = new Hashids(HASH_SALT, MIN_LENGTH)

/**
 * Decodes a string id into an int. Returns null if an invalid ID.
 */
export const decodeHashId = (id: string) => {
  try {
    const ids = hashids.decode(id)
    if (!ids.length) return null
    const num = Number(ids[0])
    if (isNaN(num)) return null
    return num
  } catch (e) {
    return null
  }
}

/**
 * Encodes an int to a string based hashid
 */
export const encodeHashId = (id: number | null) => {
  try {
    if (id === null) return null
    const encodedId = hashids.encode(id)
    return encodedId
  } catch (e) {
    return null
  }
}
